### ISSUE TITLE
Cached Interpreter- Game Freeze Sonic Adventure 2

### ISSUE THREAD

**Issue by brand175** [OatmealDome/dolphin-ios#173]
Game freezes in-game around 20 seconds. Game runs fine on JITARM64. Using single core or disabling OC doesn't do anything to fix it
![IMG_1260](https://github.com/user-attachments/assets/7feb9b5e-6eea-4828-9b01-f979c889f5ff)

**Comment by OatmealDome** [comment]
Are you able to test to see if this happens on a PC or Mac? If it happens on official Dolphin, I can forward the bug to them.

**Comment by brand175** [comment]
PC doesn't freeze, Android doesn't freeze. I tried using the null backend to see if it was Metal, but it still freezes

**Comment by powerhousepr** [comment]
Yep, I am testing this on iPad Pro M4 with DolphinIOS. The Shadow Story plays perfectly fine. But the Sonic Story line freezes in City Escape the same place he mentioned.

I have disabled everything and still freezes there. This is definitely a bug from the game with the emulator. Been trying to find solutions but so far nothing.

If Someone has a SAVE FILE with the story completed then send me a link. At least that way I can select levels and skip to the next one.
